src/react-medium-image-zoom.js

```diff
@@ -266,16 +274,19 @@ class Zoom extends Component {
   }
 
   render() {
+    const { defaultStyles, hasAlreadyLoaded, zoomImage } = this.props
+    const { isZoomed, src } = this.state
+
     return (
       <div onClick={ this.handleUnzoom } style={ this.getZoomContainerStyle() }>
         <Overlay
-          isVisible={ this.state.isZoomed }
-          defaultStyles={ this.props.defaultStyles }
+          isVisible={ isZoomed }
+          defaultStyles={ defaultStyles }
         />
         <img
-          src={ this.state.src }
-          alt={ this.props.alt }
-          className={ this.props.className }
+          { ...zoomImage }
+          src={ src }
+          srcSet={ hasAlreadyLoaded ? zoomImage.srcSet : null }
```

Comment:
Not backwards, but also probably unnecessarily complicated. The logic was that if src is omitted (which works for browsers supporting srcset), without this check the browser will immediately replace the currentSrc with the srcset file of its choosing, thus destroying the loading effect originally in place (where the low res image remains in place until the hi res src loads).

Though, on review I would say that should just get removed